lib/push.py

```diff
@@ -114,6 +114,15 @@ def prepare(self):
                                      "digest": "sha256:" + hash_c })
       # Prepare metadata.
       ch.INFO("preparing metadata")
+      self.image.metadata_load()
+      hist = self.image.metadata["history"]
+      for i in range(len(hist)):
+         hist[i]["empty_layer"] = True
+      try:
+         hist[-1].pop("empty_layer")
+      except IndexError:
+         ch.FATAL("metadata: missing history")
+      config["history"] = hist
```

Comment:
Would be good I think to document what we're doing with history. For example, it looks like we're using the OCI format. Why? What alternatives were considered and rejected?